clang/lib/CIR/Lowering/DirectToLLVM/LowerToLLVM.cpp

```diff
@@ -1877,6 +1877,16 @@ class CIRGlobalOpLowering
 public:
   using OpConversionPattern<mlir::cir::GlobalOp>::OpConversionPattern;
 
+  // Get addrspace by converting a pointer type.
+  // TODO: Use a more direct way to get the addrspace.
```

Comment:
What does this TODO really mean? If someone is to change this at some point it's not really clear what direct here means. Please clarify in the comment what concrete action you'd be expecting someone to later tackle.

Reply:
Actually I don't know a better solution, either. Just wanted to mark the method a little tricky. I can also remove the TODO if the approach is okay and not considered becoming a bad precedent for other part of this file.

Reply:
Updated it with a rough hint, but it should be clear enough ; )

Reply:
Comment on the big picture: It sounds a bit odd that we have to annotate both the global and the pointer though, is that really necessary? Why not just annotate the pointer and not have the global being annotated with yet another keyword? I'm a bit confused with your design here.

Reply:
Then the AS info resides on the result type of the users of `cir.global`, e.g. `cir.get_global`. It would make lowering to LLVM much harder. LLVM uses `llvm.global` with a dedicated AS attribute.

Thus, when lowering to it, we have to find every user and fetch the common one. Maybe a verification on uses, and proper consideration on how to deal with a global with no use (let's say a global just for extern) are required.

Reply:
Ok, gonna land this. Please make sure you add some verification in followup PRs and enhance the docs to explain what one should assume. Thanks!